drgn/helpers/linux/mm.py

```diff
@@ -1595,6 +1598,84 @@ def vma_find(mm: Object, addr: IntegerLike) -> Object:
         return cast("struct vm_area_struct *", mtree_load(mt, addr))
 
 
+def vma_name(vma: Object) -> str:
+    """
+    Get the display name or file path for a Virtual Memory Area (VMA).
+
+    This helper returns a human-readable label for a given
+    ``struct vm_area_struct *`` object, describing what type of memory region it represents.
+    It mimics the VMA naming behavior seen in `/proc/<pid>/maps`.
+
+    Parameters
+    ----------
+    vma : drgn.Object
+        A ``struct vm_area_struct *`` representing a virtual memory area.
+        The caller typically obtains this from the target process's ``mm_struct``.
+
+    Returns
+    -------
+    str
+        A descriptive name or file path of the VMA region.
+        Possible return values include:
+
+        - **File path** (e.g., ``/usr/lib/libc.so.6``):
+          For VMAs backed by files (`vma->vm_file` non-NULL).
+        - **"[heap]"** :
+          Heap segment, between `mm->start_brk` and `mm->brk`.
+        - **"[stack]"** :
+          Stack segment containing `mm->start_stack`.
+        - **"[vdso]"** :
+          Virtual Dynamic Shared Object area (`mm->context.vdso`).
+        - **"[vvar]"**, **"[vsyscall]"**, etc.:
+          Architecture-specific special mappings from `vm_ops->name`.
+        - **"[anon:<name>]"** :
+          Kernel versions with `anon_vma_name` support.
+    """
+
+    prog = vma.prog_
+
+    vm_file = vma.vm_file.read_()
+    if vm_file:
+        return os.fsdecode(d_path(vm_file.f_path))
+
+    vm_ops = vma.vm_ops.read_()
+    if vm_ops:
+        vm_ops_name = vm_ops.name.read_()
+        if vm_ops_name == prog["special_mapping_name"]:
+            return (
+                cast("struct vm_special_mapping *", vma.vm_private_data)
+                .name.string_()
+                .decode()
+            )
+
+    mm = vma.vm_mm.read_() if vma.vm_mm else None
+    if not mm:
+        return "[vdso]"
+
+    start = vma.vm_start.value_()
+    end = vma.vm_end.value_()
+
+    if (
+        mm.start_brk
+        and mm.brk
+        and start <= mm.brk.value_()
+        and end >= mm.start_brk.value_()
+    ):
+        return "[heap]"
+
+    if mm.start_stack and start <= mm.start_stack.value_() <= end:
+        return "[stack]"
+
+    try:
+        anon_name = mm.anon_vma_name.read_()
+        if anon_name and anon_name.name:
+            return f"[anon:{os.fsdecode(anon_name.name.string_())}]"
+    except AttributeError:
+        pass
```

Comment:
A few things here:

1. The anonymous name is `vma->anon_name`, not `mm->anon_vma_name`.
2. `anon_name->name` is a flexible array, which can't be `NULL`, so we don't need to check that.
3. We should limit the scope of the `try` block to only the `vma.anon_name` access so that we don't mask any other failed member accesses (in case members get renamed in the future, for example).
4. We should document the reason that we need the check.

```suggestion
    # anon_name was added in Linux kernel commit 9a10064f5625 ("mm: add a field
    # to store names for private anonymous memory") (in v5.17). It also only
    # exists if CONFIG_ANON_VMA_NAME=y.
    try:
        anon_name = vma.anon_name.read_()
    except AttributeError:
        pass
    else:
        if anon_name:
            return f"[anon:{os.fsdecode(anon_name.name.string_())}]"
```

I'll work on enabling `CONFIG_ANON_VMA_NAME` in drgn's vmtest kernels so we can test this codepath, but that doesn't block landing this.